Source/WebCore/platform/audio/cocoa/PlatformRawAudioDataCocoa.cpp

```diff
@@ -251,13 +251,19 @@ void PlatformRawAudioData::copyTo(std::span<uint8_t> destination, AudioSampleFor
     bool destinationIsInterleaved = isAudioSampleFormatInterleaved(destinationFormat);
 
     if (audioSampleElementFormat(sourceFormat) == audioSampleElementFormat(destinationFormat)) {
-        if (numberOfChannels() == 1 || (audioData.isInterleaved() && destinationIsInterleaved)) {
-            // Simplest case.
-            ASSERT(!planeIndex);
-            auto source = sourceList.bufferAsSpan(0);
+        bool noneInterleaved = !audioData.isInterleaved() && !destinationIsInterleaved;
+        bool bothInterleaved = audioData.isInterleaved() && destinationIsInterleaved;
+
+        // Use memcpy when both source and destination are planar.
+        // Additionally, when both are interleaved and number of
+        // channels is 1, the data layout is the same as planar.
```

Comment:
TBH I'm a little bit unsure what would it mean to treat the destination as interleaved because AFAIK the destination is an `ArrayBuffer` or an `ArrayBufferView`. So there's no information what is the amount of channels in the destination and how much distance to put between samples on the same channel.

This is coming from limited reading of the spec but perhaps both `f32-planar` and `f32` destinations are in practice equivalent?

What it would affect is the relaxing check vs the `destinationIsInterleaved` but I would like gain some confidence my understanding is correct first.

I'll retest some a few things when the current build completes :bow: